### PR TITLE
Add AppStream Generator

### PR DESCRIPTION
**Pull request by ximion**
Hi everyone!

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

This adds a Flatpak manifest for the [AppStream Generator](https://github.com/ximion/appstream-generator) tool, since it's not super trivial to build and something people frequently want to upgrade on stable distribution. So far there was a Snap package available as well, but it makes sense to also have this available as Flatpak for people who prefer Flatpak bundles.

Thanks for considering!
    Matthias

### CODE REVIEW

File: org.freedesktop.appstream.generator.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  subdir: libraries/liblmdb
+  cleanup:
+   - /share/man/man1/*
+   - /bin/*
+  sources:
+   - type: archive
+     url: https://git.openldap.org/openldap/openldap/-/archive/LMDB_0.9.30/openldap-LMDB_0.9.30.tar.bz2
+     sha256: eb16ed6fd535b85857c331b93e7f9fd790bc9fcea3fa26162befdc1ba2775668
+
+- name: lmdb.pc
```
Comment (hfiguiere) [inline org.freedesktop.appstream.generator.yaml:94]: I'd just merge this module with `lmdb`.

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.appstream.generator

File: org.freedesktop.appstream.generator.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
+---
+app-id: org.freedesktop.appstream.generator
```
Comment (hfiguiere) [inline org.freedesktop.appstream.generator.yaml:2]: not completely sure of the policy to use `org.freedesktop` here.

Comment (ximion) [reply org.freedesktop.appstream.generator.yaml:2]: Well, this is a Freedesktop project and I am the upstream maintainer of AppStream, so I am sure it's okay in this case ;-)

File: org.freedesktop.appstream.generator.yaml (hunk 1 above)
Comment (ximion) [reply org.freedesktop.appstream.generator.yaml]: Done :-)

**Comment by ximion** [comment]
The build failure should be fixed with the latest change :-)

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.appstream.generator

**Comment by ximion** [comment]
Thank you for the review & merge! :-D

**Pull request closed without merging**